noir-projects/aztec-nr/aztec/src/state_vars/private_set.nr

```diff
@@ -275,8 +275,11 @@ where
     where
         Note: Packable<N = M>,
     {
-        let (retrieved_notes, note_hashes) =
-            get_notes(self.context, self.owner, self.storage_slot, options);
+        std::static_assert(options.owner.is_none(), "Owner cannot be set in pop_notes");
```

Comment:
This is now a bit of an API regression as the input arg types allow setting of owner while it's incorrect to do here as it's defined by the state variable.

But this ugliness is fairly self-contained and should be always possible to figure out at comptime (Noir should see that we never attempt at setting it anywhere) and hence should not introduce constraints.